content/blog/introducao-orientacao-objetos.md

```diff
@@ -0,0 +1,381 @@
+
+### 1. Abstração
+Em uma definição própria:
+> Abstração é a capacidade de delimitar um objeto, suas propriedades e comportamentos, ou seja, o que o define, de forma que seja uma representação fiel à realidade ao qual está inserido.
+
+Do site [dicio](https://www.dicio.com.br/abstracao/), com grifo próprio
+> Operação do espírito, que isola de uma noção um elemento, negligenciando os outros.
+
+Chegamos a importante conclusão: Para OOP devemos definir de forma muita clara o que o objeto representa
+
+Um sistema desenvolvido utilizando o paradigma orientado a objetos é formado por estruturas que representam os objetos da vida real, que estarão presentes no sistema, representados através dessas abstrações.
+
+Uma abstração define as propriedades, comportamentos e os eventos ocorridos a um objeto.
+
+#### Exemplo
+
+Um exemplo de abstração pode ser aplicado a um automóvel, Digamos que tenhamos uma classe Automóvel, quais são as características que o carro tem?
+
+Podemos enumerar várias:
+
+-  Fabricante; Marca; Modelo; Ano de fabricação
+-  Distância entre eixos; Altura; Largura
+
+É um erro comum descrever todas as propriedades que compõem o objeto, contudo a abstração é um exercício, devemos nos ater àquilo que é importante para o domínio da aplicação, e podemos e devemos progredir nesse exercício.
+
+> O Domínio da aplicação é delimitado pelo problema que ela visa resolver
+
+```csharp
+Class Automovel {
+ DateTime DataDeFabricacao;
+ string CodigoDeReferenciaDoFabricante;
+ string Modelo;
+
+ float Velocidade;
+ Acelerar(){ }
+ Frear(){ }
+
+ Colidiu(){ }
+ }
+```
+<small>Listagem 1: Classe automóvel</small>
+
+Dessa forma escrevemos uma classe Automovel que tem como propriedades DataDeFabricacao, CodigoDeReferenciaDoFabricante e Modelo, ela também apresenta os comportamentos para Acelerar e Frear. 
+
+Note que existe um comportamento representado pela assinatura Colidiu, contudo colidir não é um comportamento do carro. O que é então? Um evento, algo que acontece com o Automovel através de uma casualidade, ou imprevisto, geralmente são as exceções.
+
+Agora que temos uma estrutura responsável por representar um automóvel no nosso domínio podemos apresentar o conceito de encapsulamento.
+
+### Encapsulamento
+
+Podemos pensar em encapsulamento na forma mais fácil de associação possível: Uma capsula de remédio.
+
+Dentro de uma capsula gelatinosa estão todos os componentes que contribuem para a melhoria, todos os ingredientes est
+
+Dicio
+Abstração
+Significado de Abstração no Dicionário Online de Português. O que é abstração: s.f. Operação do espírito, que isola de uma noção um elemento, negligen...
+ ão contidos dentro da capsula.
+
+Encapsulamento em OOP é nada mais que definir uma API (Application Interface) pública através de uma abstração que entrega o comportamento especificado ao que consome tal API.
+
```

Comment:
Não creio que esta explicação está correta.
Definição de encapsulamento em inglês:
Encapsulation is one of the fundamentals of OOP (object-oriented programming). It refers to the bundling of data with the methods that operate on that data.[5] Encapsulation is used to hide the values or state of a structured data object inside a class, preventing unauthorized parties' direct access to them. Publicly accessible methods are generally provided in the class (so-called getters and setters) to access the values, and other client classes call these methods to retrieve and modify the values within the object.